src/tqec/templates/tiled.py

```diff
@@ -0,0 +1,163 @@
+from copy import deepcopy
+from typing import Sequence
+
+import numpy
+import numpy.typing as npt
+from typing_extensions import override
+
+from tqec.exceptions import TQECException
+from tqec.position import Displacement, Position2D, Shape2D
+from tqec.templates.base import RectangularTemplate
+from tqec.templates.enums import TemplateSide
+from tqec.templates.scale import Scalable2D
+
+
+class TiledTemplate(RectangularTemplate):
+    def __init__(
+        self,
+        template_by_position: dict[Position2D, RectangularTemplate],
+        k: int = 2,
+        default_increments: Displacement | None = None,
+    ) -> None:
+        """A template representing a tiling of other templates.
+
+        The tiles are indexed by their position in the tiling.
+        """
+        super().__init__(k, default_increments)
+        if not template_by_position:
+            raise TQECException(
+                "Cannot create a tiled template with an empty template map."
+            )
+
+        scalable_shapes = {
+            template.scalable_shape for template in template_by_position.values()
+        }
+        if len(scalable_shapes) != 1:
+            raise TQECException(
+                "All templates in the TiledTemplate should have the same scalable shape."
+            )
+        self._base_scalable_shape = scalable_shapes.pop()
+
+        # Build plaquette indices map
+        indices_map: dict[Position2D, dict[int, int]] = {}
+        index_count = 1
+        indices_map = {}
+        for position, template in template_by_position.items():
+            indices_map[position] = {
+                i: i + index_count for i in range(template.expected_plaquettes_number)
+            }
+            index_count += template.expected_plaquettes_number
+        self._indices_map = indices_map
+
```

Comment:
`self._indices_map` does not seem to be used anywhere. Did I miss something?

Reply:
Good catch! I forgot to remove that one.